### ISSUE TITLE
Cannot exec on a running container: oci runtime error: exec failed: permission denied

### ISSUE THREAD

**Issue by chefarov** [docker/docker#22199]
**Output of `docker version`:**

```
Client:
 Version:      1.11.0
 API version:  1.23
 Go version:   go1.5.4
 Git commit:   4dc5990
 Built:        Wed Apr 13 18:34:23 2016
 OS/Arch:      linux/amd64

Server:
 Version:      1.11.0
 API version:  1.23
 Go version:   go1.5.4
 Git commit:   4dc5990
 Built:        Wed Apr 13 18:34:23 2016
 OS/Arch:      linux/amd64
```

**Output of `docker info`:**

```
Containers: 5
 Running: 0
 Paused: 0
 Stopped: 5
Images: 139
Server Version: 1.11.0
Storage Driver: aufs
 Root Dir: /var/lib/docker/aufs
 Backing Filesystem: extfs
 Dirs: 134
 Dirperm1 Supported: true
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins: 
 Volume: local
 Network: host bridge null
Kernel Version: 3.19.0-32-generic
Operating System: Ubuntu 14.04.4 LTS
OSType: linux
Architecture: x86_64
CPUs: 8
Total Memory: 15.6 GiB
Name: barberakis
ID: GQBH:SXJO:TKYE:2ITM:WHV2:GGAQ:AVOI:CHYT:7V3K:MGNN:L57X:7JAC
Docker Root Dir: /var/lib/docker
Debug mode (client): false
Debug mode (server): false
Registry: https://index.docker.io/v1/
WARNING: No swap limit support
```

**Additional environment details (AWS, VirtualBox, physical, etc.):**

Physical workstation (Linux Mint):
$ `uname -a: Linux hostname 3.19.0-32-generic #37~14.04.1-Ubuntu SMP Thu Oct 22 09:41:40 UTC 2015 x86_64 x86_64 x86_64 GNU/Linux`

**Steps to reproduce the issue:**
1. `sudo docker-compose up`
2. `sudo docker exec -u root -ti my_container_name /bin/bash`

**Describe the results you received:**

(2) gives: 
`rpc error: code = 2 desc = "oci runtime error: exec failed: permission denied"`

**Describe the results you expected:**

A bash shell inside the running container

**Additional information you deem important (e.g. issue happens only occasionally):**

- `docker attach` works
- `CTRL-C/Z` on the running container or
- `docker rm my_container_name` or 
- `docker kill my_container_name` , all throw: 
`Error response from daemon: Cannot kill container aa: Cannot kill container db76a115cf6cb33bea09396b816b5826628d184709d663466f82f466fc813e4c: rpc error: code = 7 desc = "permission denied"`

Restarting the daemon kills the container successfully. Running up again the container with 
`sudo docker-compose up`, 
I get:
` ERROR: for my_container_name  rpc error: code = 6 desc = "mkdir /run/containerd/db76a115cf6cb33bea09396b816b5826628d184709d663466f82f466fc813e4c: file exists"`, which is [#21702](https://github.com/docker/docker/issues/21702)  

This is solved with 
`rm -rf /run/containerd/db76a115cf6cb33bea09396b816b5826628d184709d663466f82f466fc813e4c &&
sudo docker rm -f my_container_name`, 

Then we start the same cycle...
`sudo docker-compose up` ...

**Comment by thaJeztah** [comment]
Just for completeness, what version of docker-compose are you using, and do you have the docker-compose file to reproduce (or preferably a simple-as-possible compose file)

**Comment by chefarov** [comment]
`docker-compose version 1.7.0, build 0d7bf73`

my file:

```
core:
    image: tomcat:8
    #restart: always
    env_file: env_file
    volumes:
        - ./client-api/target/rgs-core.war:/usr/local/webapps/rgs-core.war
    ports:
        - "9999:8080"
    container_name: "aa"
```

Thanks

**Comment by mlaventure** [comment]
@chefarov I've tried to reproduce your error with your compose file (minus the volumes and env_file directive) and had no luck so far.

Could we try reproducing after having followed the step in https://github.com/docker/docker/issues/21702#issuecomment-210224160 ?

**Comment by chefarov** [comment]
I cannot reproduce it either, using same (kernel, docker, docker compose) versions on another PC.

**Comment by chefarov** [comment]
Substituting docker-runc didn't make any difference.

More crazy stuff. I dont know if this helps somehow, but I thought I should report it as well.
If I change the `docker-compose.yml` file to:
```
vrgs:
    build: .
    #image: tomcat:8.0-jre8
    env_file: my_env
    ports:
        - "9999:8080"
    container_name: devil
```

with `Dockerfile`:
```
FROM tomcat:8.0-jre8

ENV $CATALINA_HOME /usr/local/tomcat/
ENV $WAR_FILE ./client-api/target/rgs-core.war

#Deploy application & configure tomcat
COPY $WAR_FILE $CATALINA_HOME/webapps/rgs-core.war
COPY ./conf/tomcat/*.xml $CATALINA_HOME/conf/
```

Then I can successfully
`sudo docker-compose build`
but:
`sudo docker-compose up` 
gives:
```
ERROR: for vrgs  rpc error: code = 2 desc = "oci runtime error: write /proc/30781/oom_score_adj: invalid argument"
Attaching to 
```

This later issue (not the previous) appears also on a second computer running ubuntu 16.04, that has same (kernel, docker, docker-compose) versions as the first machine (docker info same stuff).

If I remove the ENVIRONMENT tags (that override the my_env file values), inside `docker-compose.yml` file:

FROM tomcat:8.0-jre8

#Deploy application & configure tomcat, nginx
COPY $WAR_FILE $CATALINA_HOME/webapps/rgs-core.war
COPY ./conf/tomcat/*.xml $CATALINA_HOME/conf/

then everything **works fine** on **the later machine only!**

I can't connect the dots of what is relevant to the initial issue and what is not. Since I cannot reproduce the issue on the second computer I can't help any more, thus you could close this if you want.

I will format the first computer at some stage, and report back if this was resolved.

**Comment by cpuguy83** [comment]
Your environment variables are a little weird `$CATALINA_HOME`, should be `CATALINA_HOME`

Also, how big is your env file?

**Comment by chefarov** [comment]
Thank you. I can't believe I made that hilarious mistake with the envs, removing the dollars solves the last problem. Sorry for the spam. 

The initial problem persists, however I imagine something peculiar is going on with the host machine. I just ran successfully on another machine (ubuntu 12.04 VM). Thus the problem can't be reproduced. We can close it, if you think so

**Comment by seffenberg-naspers** [comment]
I have the same issue (docker v1.11.1, ubuntu 16.04) but what is interesting:

It is a jenkins slave.. it had only 1 executor and I was starting a bunch of containers on each job. This worked fine.
As soon as I added 2 more executors and tried to start the same kind of containers (same base images, only named differently) multiple times I got this error or the error described in #22226 which might be connected.

**Comment by mlaventure** [comment]
@seffenberg-naspers is most likely a different issue

Could you please create a new issue with more details regarding what you're seeing (and also your specific configuration and relevant docker daemon logs) and if possible a way to reproduce it.

I'll close this issue meanwhile.

Thanks.

**Comment by xiaods** [comment]
i came into below issue 

```
docker start omega-slave 
Error response from daemon: rpc error: code = 6 desc = "mkdir /run/containerd/1b375446988d74337d3dc7865553363ea599bf622a70255e2d7f3f47d06ef544: file exists
```

in docker 1.11.2

```
[root@localhost ~]# docker info
Containers: 10
 Running: 10
 Paused: 0
 Stopped: 0
Images: 17
Server Version: 1.11.2
Storage Driver: devicemapper
 Pool Name: docker-253:0-394838-pool
 Pool Blocksize: 65.54 kB
 Base Device Size: 107.4 GB
 Backing Filesystem: xfs
 Data file: /dev/loop0
 Metadata file: /dev/loop1
 Data Space Used: 3.577 GB
 Data Space Total: 107.4 GB
 Data Space Available: 35.07 GB
 Metadata Space Used: 5.861 MB
 Metadata Space Total: 2.147 GB
 Metadata Space Available: 2.142 GB
 Udev Sync Supported: true
 Deferred Removal Enabled: false
 Deferred Deletion Enabled: false
 Deferred Deleted Device Count: 0
 Data loop file: /var/lib/docker/devicemapper/devicemapper/data
 WARNING: Usage of loopback devices is strongly discouraged for production use. Either use `--storage-opt dm.thinpooldev` or use `--storag
e-opt dm.no_warn_on_loop_devices=true` to suppress this warning.
 Metadata loop file: /var/lib/docker/devicemapper/devicemapper/metadata
 Library Version: 1.02.107-RHEL7 (2015-12-01)
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins: 
 Volume: local
 Network: bridge null host
Kernel Version: 3.10.0-327.13.1.el7.x86_64
Operating System: CentOS Linux 7 (Core)
```

**Comment by thaJeztah** [comment]
@xiaods looks like that's issue https://github.com/docker/docker/issues/21702
